File: app/src/main/java/com/example/taptalkuisampleapp/SampleApplication.java
package com.example.taptalkuisampleapp;

import android.app.Application;
import android.util.Log;

import io.taptalk.TapTalk.Helper.TapTalk;
import io.taptalk.TapTalk.Listener.TapListener;
import io.taptalk.TapTalk.Model.TAPMessageModel;

public class SampleApplication extends Application {

    private String TAG = getClass().getSimpleName();

    @Override
    public void onCreate() {
        super.onCreate();
        TapTalk.init(this, "d1e5dfe23d1e00bf54bc2316f",
                "NTQzMTBjZDI5YWNjNTEuMS4x/ZDY4MTg3Yjg/OTA0MTQwNDFhMDYw/MGI0YjA5NTJjM2Fh",
                //R.mipmap.ic_launcher, getResources().getString(R.string.tap_app_name),
                R.mipmap.ic_launcher, "TapTalk Dev Sample App", "engine-dev.taptalk.io",
                TapTalk.TapTalkImplementationType.TapTalkImplementationTypeUI,
                tapListener);
        TapTalk.initializeGooglePlacesApiKey("AIzaSyA1kCb7yq2shvC3BnzriJLcTfzQdmzSnPA");
    }

    private TapListener tapListener = new TapListener() {
        @Override
        public void onTapTalkRefreshTokenExpired() {
            super.onTapTalkRefreshTokenExpired();
            Log.e(TAG, "onTapTalkRefreshTokenExpired: ");
        }

        @Override
        public void onTapTalkUnreadChatRoomBadgeCountUpdated(int i) {
            super.onTapTalkUnreadChatRoomBadgeCountUpdated(i);
            Log.e(TAG, "onTapTalkUnreadChatRoomBadgeCountUpdated: ");
        }

        @Override
        public void onNotificationReceived(TAPMessageModel tapMessageModel) {
            super.onNotificationReceived(tapMessageModel);
            Log.e(TAG, "onNotificationReceived: ");
        }
    };
}

File: app/src/main/java/com/example/taptalkuisampleapp/SampleApplication.kt
package com.example.taptalkuisampleapp

import android.app.Application
import android.content.Intent
import io.taptalk.TapTalk.Helper.TapTalk
import io.taptalk.TapTalk.Listener.TapListener
import io.taptalk.TapTalk.Model.TAPMessageModel

class SampleApplication : Application() {

    private val TAG = javaClass.simpleName

    private val tapListener = object : TapListener() {
        override fun onTapTalkRefreshTokenExpired() {
            super.onTapTalkRefreshTokenExpired()
            val intent = Intent(applicationContext, TAPLoginActivity::class.java)
            intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
            applicationContext.startActivity(intent)
        }

        override fun onTapTalkUnreadChatRoomBadgeCountUpdated(i: Int) {
            super.onTapTalkUnreadChatRoomBadgeCountUpdated(i)
        }

        override fun onNotificationReceived(tapMessageModel: TAPMessageModel?) {
            super.onNotificationReceived(tapMessageModel)
            TapTalk.showTaptalkNotification(tapMessageModel)
        }

        override fun onUserLogout() {
            super.onUserLogout()
        }
    }

    override fun onCreate() {
        super.onCreate()
        TapTalk.setLoggingEnabled(true)
        TapTalk.init(
            this, "APP_KEY_ID",
            "APP_KEY_SECRET",
            R.mipmap.ic_launcher, "TapTalk Dev Sample App",
            "API_BASE_URL",
            TapTalk.TapTalkImplementationType.TapTalkImplementationTypeUI,
            tapListener
        )
        TapTalk.initializeGooglePlacesApiKey("GOOGLE_PLACES_API_KEY")
    }
}
